tests/conftest.py

```diff
@@ -121,16 +193,127 @@ def hardened(request):
 @pytest.fixture
 def _file(request):
     """Use a local file directory for root-dir instead of FTP or HTTP."""
-    return request.config.getoption("--file")
+    return request.config.getoption('--file')
 
 
 @pytest.fixture
 def auth(request):
     """Authentication method for HTTP fetching. Valid values: basic, digest"""
-    return request.config.getoption("--auth")
+    return request.config.getoption('--auth')
 
 
 @pytest.fixture
 def noupdate(request):
     """ Decide whether or not to update the fetch to the latest patch level."""
-    return request.config.getoption("--noupdate")
+    return request.config.getoption('--noupdate')
+
+
+@pytest.fixture(scope='session')
+def jails():
+    return {}
+
+
+@pytest.fixture
+def config_data():
+    def read_file(jail):
+        jails = iocage.IOCage(jail=jail).jails
+        if (
+            jail in jails and os.path.exists(
+                os.path.join(jails[jail], 'config.json')
+            )
+        ):
+            with open(os.path.join(jails[jail], 'config.json'), 'r') as f:
+                return json.loads(f.read())
+        else:
+            return None
+
+    return read_file
+
+
+@pytest.fixture
+def invoke_cli():
+    def invoke(cmd, reason=None):
+        runner = CliRunner()
+
+        result = runner.invoke(
+            ioc.cli, cmd
+        )
+
+        reason = f'{reason}: {result}' if reason else result
+
+        assert result.exit_code == 0, reason
+
+        return result
+
+    return invoke
+
+
+@pytest.fixture
+def run_cmd():
+    def run(cmd):
+        # Going to raise exception if cmd returns non zero status
+        return iocage_lib.ioc_common.checkoutput(cmd, stderr=subprocess.PIPE)
+
+    return run
+
+
+@pytest.fixture
+def write_file():
+    def write_to_file(location, data):
+        with iocage_lib.ioc_common.open_atomic(location, 'w') as f:
+            f.write(data)
+
+    return write_to_file
+
+
+@pytest.fixture
+def remove_file():
+    def remove(path):
+        if os.path.exists(path):
+            os.remove(path)
+
+    return remove
+
+
+@pytest.fixture(scope='session')
+def thin_jails(jails):
+    return {k: v for k, v in jails.items() if v.get('type', 'thin') == 'thin'}
+
+
+@pytest.fixture(scope='session')
+def thick_jails(jails):
+    return {k: v for k, v in jails.items() if v.get('type', 'thin') == 'thick'}
+
+
+@pytest.fixture(scope='session')
+def thick_config_jails(jails):
```

Comment:
We don't need these 3 classes. You're already reading the configuration data, so in the test that wants to verify that it's a thick jail configuration, just look for the CONFIG_TYPE key, otherwise it's thin.